### PR TITLE
sci-physics/qutip: new ebuild

### PR DESCRIPTION
**Pull request by lorem-ipsum**
Qutip is a python library for quantum simulation. I took the liberty to add the science project as a maintainer, but so far I have no response from the project whether that is desired. 

Package-Manager: Portage-2.3.5, Repoman-2.3.2

### CODE REVIEW

File: sci-physics/qutip/qutip-4.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	dev-python/matplotlib[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+"
+DEPEND="${RDEPEND}
+	test? ( dev-python/nose[${PYTHON_USEDEP}] )
+"
+
+pkg_pretend() {
+	use openmp && tc-check-openmp
+}
```
Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: handling OpenMP is a bit more tricky:

```
pkg_pretend() {
	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
}

pkg_setup() {
	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
}
```
Yes, you *need* to check for it in `pkg_setup`, and you *may* check for it in `pkg_pretend`

File: sci-physics/qutip/qutip-4.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{5,6} )
+inherit distutils-r1 toolchain-funcs
```
Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: empty line before... and why no py3.4?

File: sci-physics/qutip/qutip-4.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+pkg_pretend() {
+	use openmp && tc-check-openmp
+}
+
+src_prepare() {
+	sed -i setup.py \
+		-e 's/_compiler_flags = .*$/_compiler_flags = []/' || die
+
+	default
+}
```
Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: don't override src_* functions for python. Rather use `python_prepare_all`, and call `distutils-r1_python_prepare_all` at the end

File: sci-physics/qutip/qutip-4.1.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+python_configure_all() {
+	if use openmp ; then
+		mydistutilsargs=( --with-openmp )
+	fi
```
Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: shorten this: `use openmp && mydistutilsargs=( --with-openmp )`

File: sci-physics/qutip/qutip-4.1.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+python_test() {
+	cd "${BUILD_DIR}"/lib* || die
+	${PYTHON} -c "import qutip.testing as qt ; qt.run()" || die
```
Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: `EPYTHON` instead of `PYTHON`

File: sci-physics/qutip/qutip-4.1.2.ebuild (hunk 2 above)
Comment (lorem-ipsum) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: Upstream recommends python3.5+, I haven't tested whether 3.4 works but will check (test suite just takes ages)

File: sci-physics/qutip/qutip-4.1.2.ebuild (hunk 1 above)
Comment (lorem-ipsum) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: Oh, ok, will adjust it accordingly. A warning in the eclass might  be helpful though.

Comment (SoapGentoo) [inline sci-physics/qutip/qutip-4.1.2.ebuild]: @lorem-ipsum nope, there's no way that could be turned into a warning, given how PMS and eclasses work.

**Pull request closed without merging**